test/server/cards/01_SOR/leaders/BobaFettCollectingTheBounty.spec.js

```diff
@@ -0,0 +1,94 @@
+describe('Boba Fett, Collecting the Bounty', function() {
+    integration(function() {
+        describe('Boba Fett\'s leader ability', function() {
+            beforeEach(function() {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['rivals-fall', 'cantina-bouncer'],
+                        groundArena: ['death-star-stormtrooper'],
+                        leader: 'boba-fett#collecting-the-bounty',
+                        base: 'dagobah-swamp',
+                        resources: 6,
+                    },
+                    player2: {
+                        groundArena: ['cell-block-guard'],
+                        leader: { card: 'luke-skywalker#faithful-friend', deployed: true },
+                    },
+                });
+            });
+
+            it('should ready a resource when an enemy unit leaves play - defeated', function() {
+                this.player1.clickCard(this.rivalsFall);
```

Comment:
For tests like this that are primarily attack-based we tend to consolidate them down to fewer tests as much as possible by un-exhausting units in between, bc Jasmine tests are extremely heavy and we have finite github actions time.

It's bad test practice but kind of a practical limitation we're trying to observe for now. Check out the Iden tests and see how that's set up as a reference.

Reply:
I don't have time to do a thorough review of tests on this pass but I think there's plenty in this review already, I'll take a detailed look at tests once the impl stuff is updated.